### PR TITLE
Refactor playingfield 3

### PR DESCRIPTION
**Pull request by Taeir**
- Fixed some GameThread issues.
- Cleaned up some code
- Fixed achievements not working with new changes
- Removed GUI code from SinglePlayerPlayingField, the SinglePlayerController now contains the code required to update the score and life displays.
- Simplified some GameThread code.

Due to the changes that I had to make in the Achievements, this pull-request has become harder to read. I advise the following order:

PlayingField
SinglePlayerPlayingFIeld
SinglePlayerController
GameThread

Entity
PlayerFish
PlayerDeathObserver

<feature,organisation,interface,documentation,user-interface

### CODE REVIEW

File: src/main/java/com/github/fishio/control/SinglePlayerController.java [hunk 1]
```diff
@@ -27,9 +28,38 @@
  * The controller class of the single player game.
  */
 public class SinglePlayerController implements ScreenController {
```
Comment (Diocruel) [inline src/main/java/com/github/fishio/control/SinglePlayerController.java:30]: _Auto-generated comment_

```
This class has too many methods, consider refactoring it.
```

Comment (Diocruel) [inline src/main/java/com/github/fishio/control/SinglePlayerController.java:30]: _Auto-generated comment_

```
The class 'SinglePlayerController' has a Cyclomatic Complexity of 2 (Highest = 11).
```

**Comment by Diocruel** [comment]
Except those two errors it looks really nice. Good use of the observer design pattern again. Not sure if it possible to fix the error, if not, the pr is good to merge.

<layout, organisation

**Comment by LMiljak** [comment]
I honestly can't think of a better solution, those two warnings mentioned by Robin will have to do. Unless someone can come up with something, I think it's good to merge.

<layout, organisation

**Comment by Taeir** [comment]
I made some changes and reduced the number of PMD warnings.

**Comment by Diocruel** [comment]
The change to the pmd file should remove a lot of warnings in other files too, I still find it an annoying check that is made, because it also reports test classes. Anyway good to merge.

<layout, organisaton

**Comment by Taeir** [comment]
@Diocruel I attempted to write my own rule for it, which excludes methods whose name starts with "test", but I had some problems getting it to work, so I decided not to.

**Pull request merged**